metaflow/plugins/pypi/conda_environment.py

```diff
@@ -269,11 +269,9 @@ def get_environment(self, step):
         # Resolve `linux-64` Conda environments if @batch or @kubernetes are in play
         target_platform = conda_platform()
         for decorator in step.decorators:
-            # TODO: rather than relying on decorator names, rely on attributes
-            #       to make them extensible.
-            if decorator.name in ["batch", "kubernetes", "nvidia", "snowpark", "slurm"]:
+            if getattr(decorator, "support_conda", None):
```

Comment:
for backward compatibility we can have this check as an `or` with the decorator names.

Reply:
added the fallback for the non-core decorators. or should batch/kubernetes also be on the list?

Reply:
added a note on backward compatibility, and kept the complete list of decorators. Motivation being that some forks/extensions might have their own implementation for batch/kubernetes decorators as well that we do not want to break.